### PR TITLE
argo workflow enhancement 

### PR DESCRIPTION
**Pull request by season1946**
### What does this PR do?
use eks blueprints to provision argo workflows
add argo events in the example and extend md 

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/awslabs/data-on-eks/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR to add a new Add-on for [Terraform EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) repo (if applicable)
- [ ] Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: schedulers/argo-workflow/addons.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module "eks_blueprints_kubernetes_addons" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.14.0"
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons"
```
Comment (vara-bonthu) [inline schedulers/argo-workflow/addons.tf:2]: Can we lock this to latest version?

File: schedulers/argo-workflow/addons.tf [hunk 2]
```diff
@@ -115,3 +100,22 @@ resource "kubernetes_role_binding" "argo-admin-rolebinding" {
...
+  create_kubernetes_namespace = true
+  kubernetes_namespace        = "argo-events"
+  kubernetes_service_account  = "event-sa"
+  irsa_iam_policies           = [data.aws_iam_policy.sqs.arn]
+  eks_cluster_id              = module.eks_blueprints.eks_cluster_id
+  eks_oidc_provider_arn       = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${module.eks_blueprints.oidc_provider}"
+}
+
+data "aws_iam_policy" "sqs" {
+
```
Comment (vara-bonthu) [inline schedulers/argo-workflow/addons.tf:119]: remove the extra space

File: schedulers/argo-workflow/addons.tf [hunk 3]
```diff
@@ -40,30 +40,15 @@ module "eks_blueprints_kubernetes_addons" {
...
-    irsa_iam_role_path             = "/"
-    irsa_iam_permissions_boundary  = ""
-  }
-}
 
 #---------------------------------------------------------------
-# Kubernetes Cluster role for argo workflows
+# Kubernetes Cluster role for argo workflows to run spark jobs
 #---------------------------------------------------------------
 resource "kubernetes_cluster_role" "spark-cluster" {
```
Comment (vara-bonthu) [inline schedulers/argo-workflow/addons.tf:53]: Resource name should use underscore (`_`). e.g.,  `spark_cluster`. Same applies for all resource names and module names.

Comment (vara-bonthu) [reply schedulers/argo-workflow/addons.tf:53]: May be change the resource name to `argo_workflows` and the metadata.name to `argo-workflows-role`

File: schedulers/argo-workflow/addons.tf [hunk 4]
```diff
@@ -82,13 +67,13 @@ resource "kubernetes_cluster_role" "spark-cluster" {
 resource "kubernetes_role_binding" "spark_role_binding" {
   metadata {
     name      = "argo-spark-rolebinding"
```
Comment (vara-bonthu) [inline schedulers/argo-workflow/addons.tf:69]: We can change this to `argo-workflows-rolebinding` for consistency

File: website/docs/job-schedulers-eks/argo-workflows-eks.md [hunk 5]
```diff
@@ -18,43 +20,45 @@ Ensure that you have the following tools installed locally:
...
 3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
-4. [Argo WorkflowCLI](https://github.com/argoproj/argo-workflows/releases/latest)
+
 
 ## Deploy
 
 To provision this example:
 
 ```bash
-git clone https://github.com/awslabs/data-on-eks.git
```
Comment (vara-bonthu) [inline website/docs/job-schedulers-eks/argo-workflows-eks.md:28]: We need got clone command as well in the steps to make it clear for users

File: website/docs/job-schedulers-eks/argo-workflows-eks.md [hunk 6]
```diff
@@ -5,9 +5,11 @@ sidebar_position: 4
 # Argo Workflows on EKS
```
Comment (vara-bonthu) [inline website/docs/job-schedulers-eks/argo-workflows-eks.md:5]: Verify the doc format by running `npm installl` and `npm start` inside website folder.

**Review by vara-bonthu — COMMENTED** [review]
Thanks @season1946 🚀  Few minor comments

File: schedulers/argo-workflow/addons.tf (hunk 1 above)
Comment (season1946) [reply schedulers/argo-workflow/addons.tf:2]: 4.14 is the latest version. my change was merged but not in 4.14

File: schedulers/argo-workflow/addons.tf (hunk 3 above)
Comment (season1946) [reply schedulers/argo-workflow/addons.tf]: changed to argo-workflows-spark-rolebinding
 and argo-workflows-admin-rolebinding

**Comment by season1946** [comment]
@vara-bonthu npm run results 
image

File: schedulers/argo-workflow/addons.tf (hunk 1 above)
Comment (vara-bonthu) [reply schedulers/argo-workflow/addons.tf:2]: Ok, change this to 4.15. Bryant will create a new release this weekend for eks blueprints. We can merge this PR once the latest tag is available

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Pull request merged**